### PR TITLE
Backport sensors system threading optimization changes

### PR DESCRIPTION
**Pull request by iche033**
# 🎉 New feature

## Summary
Backport changes from #1938 and #2025

## Test it

* Run the `INTEGRATION_sensors_system_update_rate` test
* Run the `sensors_demo.sdf` world to make sure sensors are working properly

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by mjcarroll — APPROVED** [review]

File: src/systems/sensors/Sensors.cc [hunk 1]
```diff
@@ -17,7 +17,6 @@
 
 #include "Sensors.hh"
 
-#include <map>
```
Comment (ahcorde) [inline src/systems/sensors/Sensors.cc:20]:
```
#include <atomic>
#include <chrono>
```

File: test/integration/sensors_system_update_rate.cc [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+  private: sim::SystemLoader sm;
+};
+
+/////////////////////////////////////////////////
+TEST_F(SensorsFixture, IGN_UTILS_TEST_DISABLED_ON_MAC(UpdateRate))
+{
+  gz::sim::ServerConfig serverConfig;
+
+  const std::string sdfFile = std::string(PROJECT_SOURCE_PATH) +
+    "/test/worlds/sensor.sdf";
```
Comment (ahcorde) [inline test/integration/sensors_system_update_rate.cc:80]: `gz::common::joinPath`

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: test/integration/sensors_system_update_rate.cc (hunk 2 above)
Comment (mjcarroll) [reply test/integration/sensors_system_update_rate.cc:80]: good catch

Comment (iche033) [reply test/integration/sensors_system_update_rate.cc]: updated. [1d762d9](https://github.com/gazebosim/gz-sim/pull/2058/commits/1d762d90da3185ffe36e01bda7e10759026705dc)

File: src/systems/sensors/Sensors.cc (hunk 1 above)
Comment (iche033) [reply src/systems/sensors/Sensors.cc:20]: added. [1d762d9](https://github.com/gazebosim/gz-sim/pull/2058/commits/1d762d90da3185ffe36e01bda7e10759026705dc)

**Review by ahcorde — APPROVED** [review]

**Pull request merged**